source/GameInterface/Services/SiegeEngines/SiegeEnginesContainerRegistry.cs

```diff
@@ -0,0 +1,25 @@
+﻿using GameInterface.Services.Registry;
+using System.Threading;
+using static TaleWorlds.CampaignSystem.Siege.SiegeEvent;
+
+namespace GameInterface.Services.SiegeEngines
+{
+    internal class SiegeEnginesContainerRegistry : RegistryBase<SiegeEnginesContainer>
+    {
+        private const string SiegeEnginesContainerIdPrefix = "CoopSiegeEnginesContainer";
+        private static int InstanceCounter = 0;
+
+        public SiegeEnginesContainerRegistry(IRegistryCollection collection) : base(collection)
+        {
+        }
+
+        public override void RegisterAll()
+        {
+        }
```

Comment:
```suggestion
        {
            foreach (var siegeEnginesContainer in Campaign.Current.SiegeEventManager.SiegeEvents.Select(siegeEvent => siegeEvent.BesiegerCamp.SiegeEngines))
                {
                    // Register
                }
        }
```

Reply:
I believe Sieges and consequently their engines and progress are instanced during gameplay, do we still implement RegisterAll() for these cases?

Reply:
Yes but when a siege engine is saved and loaded it has to be registered on load which is what the RegisterAll is responsible for. Not sure if this is a special case though